subsys/bluetooth/host/conn.c

```diff
@@ -372,28 +372,32 @@ static struct bt_conn_tx *conn_tx_alloc(void)
 int bt_conn_send_cb(struct bt_conn *conn, struct net_buf *buf,
 		    bt_conn_tx_cb_t cb, void *user_data)
 {
+	int err = 0;
 	struct bt_conn_tx *tx;
 
 	BT_DBG("conn handle %u buf len %u cb %p user_data %p", conn->handle,
 	       buf->len, cb, user_data);
 
 	if (conn->state != BT_CONN_CONNECTED) {
 		BT_ERR("not connected!");
-		return -ENOTCONN;
+		err = -ENOTCONN;
+		goto end;
 	}
 
 	if (cb) {
 		tx = conn_tx_alloc();
 		if (!tx) {
```

Comment:
@Thalley @jhedberg I think it's still possible to return failure here. For example, call write command or notify in syswork. What happens when free TX space is insufficient?

``` C
static struct bt_conn_tx conn_tx[CONFIG_BT_CONN_TX_MAX];
```

Reply:
How do you mean?

We currently already return `ENOBUFS` if there are no TX contexts left.

Reply:
But in this case, the buffer is still occupied without unref.

Reply:
We lack the consideration of unref under abnormal circumstances, which is the purpose of my pr.